### PR TITLE
Changing signature of cubic spline resampling cython routine (_csplin…

### PR DESCRIPTION
**Pull request by alexis-roche**
…e_resample3d) for consistency with corresponding _cspline_sample routines. Casting of resampled arrays is now done in python.

### CODE REVIEW

**Comment by alexis-roche** [comment]
@matthew-brett Are you ok with the suggested changes?

File: nipy/algorithms/registration/resample.py [hunk 1]
```diff
@@ -16,6 +17,20 @@
...
+
+    dtype : dtype
+      Desired dtype
+    """
+    if dtype.kind in 'iu':
+        mn, mx = shared_range(arr.dtype, dtype)
+        return np.clip(np.round(arr), mn, mx).astype(dtype)
+    else:
+        return arr.astype(dtype)
+
```
Comment (matthew-brett) [inline nipy/algorithms/registration/resample.py:33]: Extra blank line for PEP8

**Comment by matthew-brett** [comment]
It would be good to have some basic tests for cast_array .

**Comment by matthew-brett** [comment]
I'll merge this, do the changes.

**Comment by matthew-brett** [comment]
Oops, sorry, I didn't notice you'd done the changes already.   I've rebased your changes on top of my branch now, thanks for that.

**Pull request closed without merging**